### PR TITLE
Added overriding styles and adding spaces functionality

### LINKED ISSUE #81: [Feature Request] Give option to a space before and after the forward slash separator
This feature request is something that it would help in being more flexible with naming conventions:
it would be nice to have a checkbox that allow the user to have or not have a space before and after the forward slash. In other words, if the user select to have spaces, the separator should be ` / `, otherwise should be `/` (current implementation).

This will allow to support both these naming conventions when, right now, only the first is supported:
image

### LINKED ISSUE #82: [Feature Request] Provide a way to export current color styles as tailwind.config.js
Once #80 is done it would be nice to be able to export the current color styles to a tailwind.config.js.
This would allow to easily modify the colors in the js file and then use the import feature to update the colors on Figma.
It would also allow to update the tailwind.config.js if, for example, some color changes are made directly in Figma.

If there are forbidden characters in the color name (e.g. spaces), those can simply be removed while exporting.

### PR DESCRIPTION
**Pull request by KA95DEV**
Added overriding styles and adding spaces functionality.

"Fixes #81 " 

"Fixes #82 "

### CODE REVIEW

**Comment by leinardi** [comment]
Nice! Thanks lot @KA95DEV! Hope to see this merged on the upstream soon :100:

File: src/code.ts [hunk 1]
```diff
@@ -2,21 +2,27 @@ import { addColors } from "./core/colors";
...
+
 interface AddColorsMessage {
   type: "ADD_COLORS";
   payload: AddColorsPayload;
 }
 
+figma.getLocalPaintStyles().forEach(style => {
+  styles.push(style)
+})
+
```
Comment (impulse) [inline src/code.ts]: Can we move the style variable down and use a map function instead?
```ts
let styles: Array<PaintStyle> = [];

styles = figma.getLocalPaintStyles().map(s => s)
```

File: src/components/ColorView.tsx [hunk 2]
```diff
@@ -27,14 +27,41 @@ function ColorView({ className }: Props) {
...
+            name="addSpaces"
+            type="checkbox"
+            checked={addSpaces}
+            onChange={() => setAddSpaces(!addSpaces)}
+          />
+          <span className="ml-2 text-gray-600 text-xs"
+          >Add spaces to slashes</span>
+        </label>
+      </div>
+
```
Comment (impulse) [inline src/components/ColorView.tsx]: The tailwind custom forms is added, we should probably style this a bit fancier like this:
```tsx
<div className="py-2">
  <label className="inline-flex items-center mt-2">
    <input
      name="overrideStyles"
      type="checkbox"
      className="form-checkbox"
      checked={overrideStyles}
      onChange={() => setOverrideStyles(!overrideStyles)}
      disabled={!hasAvailableColorStyles()}
    />
    <span className="ml-2 text-xs text-gray-600 select-none">
      Override styles if available
    </span>
  </label>

  <label className="inline-flex items-center mt-2">
    <input
      name="addSpaces"
      type="checkbox"
      className="form-checkbox"
      checked={addSpaces}
      onChange={() => setAddSpaces(!addSpaces)}
    />
    <span className="ml-2 text-xs text-gray-600 select-none">
      Add spaces to slashes
    </span>
  </label>
</div>
```
We would also need to extend the tailwind config so the style looks a bit better

`tailwind.config.js`
```js
module.exports = {
  theme: {
    customForms: theme => ({
      default: {
        input: {
          borderRadius: theme("borderRadius.default"),
          backgroundColor: theme("colors.white"),
          borderColor: theme("colors.gray.300"),
          "&:focus": {
            borderColor: theme("colors.gray.500")
          }
        },
        checkbox: {
          color: theme("colors.teal.500"),
          "&:focus": {
            borderColor: theme("colors.gray.500"),
            boxShadow: undefined
          }
        }
      }
    }),
    extend: {}
  },
  variants: {},
  plugins: [require("@tailwindcss/custom-forms")]
};
```

**Comment by KA95DEV** [comment]
> Thank you so much! Would you mind adding my suggestions?

Already applied 👍

**Comment by impulse** [comment]
Thanks, awesome!
@all-contributors please add @KA95DEV for code

**Pull request merged**